src/components/NewAppModal.js

```diff
@@ -0,0 +1,161 @@
+  headerText: { fontSize: 16, fontWeight: 600 }
+}
+
+// TODO: Once we have more appTypes, abstract text/component to support other apps
+export const NewAppModal = _.flow(
+  Utils.withDisplayName('NewAppModal'),
+  withModalDrawer({ width: 675 })
+)(({ onDismiss, onSuccess, namespace, apps, bucketName, workspaceName }) => {
+  const [viewMode, setViewMode] = useState(undefined)
+
+  const app = currentApp(apps)
+
+  const createApp = withErrorReporting('Error creating app', async () => {
+    await Ajax().Apps.app(namespace, Utils.generateKubernetesClusterName()).create(Utils.generatePersistentDiskName(), 'GALAXY', namespace, bucketName, workspaceName)
+    return onSuccess()
+  })
+
+  const deleteApp = withErrorReporting('Error deleting app', async () => {
+    await Ajax().Apps.app(app.googleProject, app.appName).delete()
+    return onSuccess()
+  })
+
+  const renderActionButton = () => {
+    return Utils.switchCase(viewMode,
+      ['deleteWarn', () => {
+        return h(ButtonPrimary, { onClick: () => deleteApp() }, ['Delete'])
+      }],
+      ['createWarn', () => {
+        return h(ButtonPrimary, { onClick: () => createApp() }, ['Create'])
+      }],
+      [Utils.DEFAULT, () => {
+        return !!app ?
+          h(ButtonPrimary, { onClick: () => setViewMode('deleteWarn') }, ['Delete']) :
+          h(ButtonPrimary, { onClick: () => setViewMode('createWarn') }, ['Next'])
+      }]
+    )
+  }
+
+  const renderBottomButtons = () => {
+    return h(Fragment, [
+      div({ style: { display: 'flex', margin: '1rem 0 1rem' } }, [
+        div({ style: { flex: 1 } }),
+        renderActionButton()
+      ])
+    ])
+  }
+
+  const getMachineDetails = () => {
+    return _.filter(({ name }) => name === 'n1-standard-8', machineTypes)[0]
+  }
+
+  const renderCreateWarning = () => {
+    return h(Fragment, [
+      div({ style: { marginBottom: '1rem' } }, ['Environment will consist of an application and cloud compute.']),
+      div({ style: { ...styles.whiteBoxContainer, backgroundColor: colors.accent(.1), boxShadow: Style.standardShadow } }, [
+        div({ style: { display: 'flex', padding: `0.5rem 0 0`, minHeight: 120 } }, [
+          div({ style: { flex: '1', lineHeight: '1.5rem', minWidth: 0, display: 'flex' } }, [
+            span({ style: { marginRight: '0.5rem', marginTop: '0.5rem' } }, [icon('info-circle', { size: 25, color: colors.accent() })]),
+            div([
+              span({ style: styles.headerText }, ['Set up duration']),
+              div({ style: { lineHeight: '24px' } }, [
+                div(['Creating a cloud environment for Galaxy takes ', span({ style: { fontWeight: 600 } }, ['8-10 minutes.'])]),
+                div(['You can navigate away, and we will notify you when it\'s ready. '])
+              ]),
+              div({ style: { ...styles.headerText, marginTop: '0.5rem' } }, ['Continuation cost']),
+              div({ style: { lineHeight: '18px' } }, [
```

Comment:
nit: These line-height values seem arbitrary and slightly different from one another. Can we standardize a bit, and/or consider using a unitless value (e.g. `1.5`) so it's relative to the text size?

Reply:
I believe they were done in that manner to try to match the mocks. However, in the name of consistency and standardized I made them just 1.5